### PR TITLE
feat(iam): ANA-4629 Update AddUserToStore docs to reflect security patch

### PR DESCRIPTION
**Pull request by mikelopez100**
# [DEVDOCS-]
Jira: [ANA-4629](https://bigcommercecloud.atlassian.net/browse/ANA-4629)

## What changed?
* Updating the AddUserToStore documentation to reflect the security patch that the IAM team tackled as part of ANA-4629. The vulnerability there was that it was possible for attackers to enumerate through user ID's and retrieve all the emails of users throughout the platform if they were successful in using an ID that belongs to someone. The IAM team recommends that we instead use email as the identifier for users as part of this specific mutation, however we are still allowing for a set of accounts that have existing Users API tokens to continue using the original API contract as before.

## Release notes draft

* 

## Anything else?

ping {names}

[ANA-4629]: https://bigcommercecloud.atlassian.net/browse/ANA-4629?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/api-docs/users/overview.mdx [hunk 1]
```diff
@@ -865,7 +865,7 @@ The following mutation creates a user:
 
 ### Add user to store
 
-The following mutation adds a user to a store:
+The following mutation adds a user to a store, the user must be part of the account:
```
Comment (exeDog) [inline docs/api-docs/users/overview.mdx:868]: In this case the input must be ID. Email skips this verification

File: docs/api-docs/users/overview.mdx [hunk 2]
```diff
@@ -876,7 +876,7 @@ The following mutation adds a user to a store:
         addUserToStore(
           input: {
             user: {
-              id: "bc/account/user/{user_id}"
+              email: "bc/account/user/{user_email}"
```
Comment (exeDog) [inline docs/api-docs/users/overview.mdx:879]: look at the above comment, you might wanna separate these 2 cases in the docs

**Review by bc-donfran — DISMISSED** [review]

File: docs/api-docs/users/overview.mdx [hunk 3]
```diff
@@ -905,6 +906,45 @@ The following mutation adds a user to a store:
...
+
+  </Tab>
+  <Tab>
+
+    ```json filename="Example mutation: Add user to store" showLineNumbers copy
+    {
+      "data": {
+        "user": {
+          "addUserToStore": {
+            "storeId": "bc/account/store/{store_hash}"
```
Comment (j05h) [inline docs/api-docs/users/overview.mdx:938]: are we really doing store hash here for `storeId` and `store_id` in the input?  ew.

Comment (mikelopez100) [reply docs/api-docs/users/overview.mdx:938]: Yes, we are indeed. It's not great

**Comment by bc-tgomez** [comment]
👋 Hey ya'll, let me know when this is ready to merge and I can give it the final review.

**Review by exeDog — APPROVED** [review]

**Comment by mikelopez100** [comment]
> 👋 Hey ya'll, let me know when this is ready to merge and I can give it the final review.

Hi @bc-tgomez 👋 this should be good for a final pass over now when you have the chance, thanks!

**Comment by bc-tgomez** [comment]
@mikelopez100 I rolled this PR into [DEVDOCS-5864](/https://bigcommercecloud.atlassian.net/browse/DEVDOCS-5864) and it's now published on main. 👍

**Pull request merged**